sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java

```diff
@@ -17,86 +22,140 @@
+        }
+        return keyFilter + originEndpoint + "/" + getLabelName(labelFilter);
     }
 
     /**
      * <p>
      * Gets settings from Azure/Cache to set as configurations. Updates the cache.
      * </p>
      * 
+     * @param trim prefix to trim
      * @throws JsonProcessingException thrown if fails to parse Json content type
      */
-    public void initProperties() throws JsonProcessingException {
-        List<String> labels = Arrays.asList(labelFilter);
-        Collections.reverse(labels);
+    public void initProperties(List<String> trim) throws JsonProcessingException {
+        if (StringUtils.hasText(snapshotName)) {
+            processConfigurationSettings(replicaClient.listSettingSnapshot(snapshotName), null, trim, true);
+        } else {
+            List<String> labels = Arrays.asList(labelFilter);
+            Collections.reverse(labels);
 
-        for (String label : labels) {
-            SettingSelector settingSelector = new SettingSelector().setKeyFilter(keyFilter + "*")
+            for (String label : labels) {
+                SettingSelector settingSelector = new SettingSelector().setKeyFilter(keyFilter + "*")
                     .setLabelFilter(label);
 
-            // * for wildcard match
-            List<ConfigurationSetting> settings = replicaClient.listSettings(settingSelector);
-
-            for (ConfigurationSetting setting : settings) {
-                String key = setting.getKey().trim().substring(keyFilter.length())
-                        .replace('/', '.');
-                if (setting instanceof SecretReferenceConfigurationSetting) {
-                    String entry = getKeyVaultEntry((SecretReferenceConfigurationSetting) setting);
-
-                    // Null in the case of failFast is false, will just skip entry.
-                    if (entry != null) {
-                        properties.put(key, entry);
-                    }
-                } else if (StringUtils.hasText(setting.getContentType())
-                        && JsonConfigurationParser.isJsonContentType(setting.getContentType())) {
-                    Map<String, Object> jsonSettings = JsonConfigurationParser.parseJsonSetting(setting);
-                    for (Entry<String, Object> jsonSetting : jsonSettings.entrySet()) {
-                        key = jsonSetting.getKey().trim().substring(keyFilter.length());
-                        properties.put(key, jsonSetting.getValue());
-                    }
-                } else {
-                    properties.put(key, setting.getValue());
+                // * for wildcard match
+                processConfigurationSettings(replicaClient.listSettings(settingSelector),
+                    settingSelector.getKeyFilter(), trim, false);
+            }
+        }
+    }
+
+    private void processConfigurationSettings(List<ConfigurationSetting> settings, String keyFilter,
+        List<String> trimStrings, boolean snapshot) throws JsonProcessingException {
```

Comment:
nit: maybe better to use `isSnapshotConfigred` or something similar instead of `snapshot`.
The `snapshot` sounds not like a boolean value.